src/library/Registrar/Adapter/Resellerclub.php

```diff
@@ -128,33 +128,25 @@ public function modifyNs(Registrar_Domain $domain)
 
     public function modifyContact(Registrar_Domain $domain)
     {
-        $customer = $this->_getCustomerDetails($domain);
-        $cdetails = $this->_getDefaultContactDetails($domain, $customer['customerid']);
-        $contact_id = $cdetails['Contact']['registrant'];
-
         $c = $domain->getContactRegistrar();
-
-        $required_params = [
-            'contact-id' => $contact_id,
-            'name' => $c->getName(),
-            'company' => $c->getCompany(),
-            'email' => $c->getEmail(),
-            'address-line-1' => $c->getAddress1(),
-            'city' => $c->getCity(),
-            'zipcode' => $c->getZip(),
-            'phone-cc' => $c->getTelCc(),
-            'phone' => $c->getTel(),
-            'country' => $c->getCountry(),
-        ];
-
-        $optional_params = [
-            'address-line-2' => $c->getAddress2(),
-            'address-line-3' => $c->getAddress3(),
-            'state' => $c->getState(),
+        $old_contact = $this -> getDomainDetails($domain);
+        /*
+        Check if the email address is the same as the one in the domain details
+        Current Fossbilling API uses the email adress of the registrant as "Username" for reseller club account
+        */
+        if($old_contact->getContactAdmin()->getEmail() != $c->getEmail()){
+            throw new Registrar_Exception('Unable to change email address');
+        }
+        $contact = $this -> getContactIdForDomain($domain);
```

Comment:
My changes to the address field don't get applied to the domain.
I changed "Street 1" to "Street 12". The information is persisted in FOSSBilling but not with the registar, it still shows "Street 1" there.

There was a new contact created but with the old data it seems:

![grafik](https://github.com/user-attachments/assets/028371f9-c45d-43cd-acd1-b43d4dd98249)

Gonna take another look tomorrow. Probably just a minor issue.

Reply:
Here is the working version of modifyContact() method that also applies changes to the contact handle. (Can't commit to this PR, that's why I'm sending it here)

```php
    public function modifyContact(Registrar_Domain $domain)
    {
        $c = $domain->getContactRegistrar();
        $old_contact = $this -> getDomainDetails($domain);
        /*
        Check if the email address is the same as the one in the domain details
        Current Fossbilling API uses the email adress of the registrant as "Username" for reseller club account
        */
        if($old_contact->getContactAdmin()->getEmail() != $c->getEmail()){
            throw new Registrar_Exception('Unable to change email address');
        }

        $customer = $this->_getCustomerDetails($domain);
        $customer_id = $customer['customerid'];
        $tld = $domain->getTld();

        //$contact = $this -> getContactIdForDomain($domain);
        [$reg_contact_id, $admin_contact_id, $tech_contact_id, $billing_contact_id] = $this->_getAllContacts($tld, $customer_id, $c);
        $params = [
            'order-id' => $this->_getDomainOrderId($domain),
            'reg-contact-id' => $reg_contact_id,
            'admin-contact-id' => $admin_contact_id,
            'tech-contact-id' => $tech_contact_id,
            'billing-contact-id' => $billing_contact_id,
        ];

        $result = $this->_makeRequest('domains/modify-contact', $params, 'POST');

        return $result['status'] == 'Success';
    }
```

Reply:
@jaapmarcus can you take a look at the revised code provided by @petzsch?

Thanks!

Reply:
Hi, @petzsch 

Would you be willing to submit a PR with your patch?
Thank you!